notify/src/lib.rs

```diff
@@ -380,6 +380,58 @@ where
     RecommendedWatcher::new(event_handler, Config::default())
 }
 
+/// Method for creating the `RecommendedWatcher` for the current platform
+/// and falling back if the recommended API is not available.
+/// 
+/// Example use case is [#423] where Docker on M1 in qemu does not expose the inotify OS API.
+///
+/// See also [`Watcher::new`](trait.Watcher.html#tymethod.new).
+pub fn recommended_watcher_fallback<F>(event_handler: F, config: Config) -> Result<WatcherFallback>
+where
+    F: EventHandler + Copy,
+{
+    // All recommended watchers currently implement `new`, so just call that.
+    match RecommendedWatcher::new(event_handler, config.clone()) {
+        Ok(v) => Ok(WatcherFallback::Native(v)),
+        Err(e) => {
+            match &e.kind {
+                #[allow(unused)]
+                ErrorKind::Io(io_error) => {
+                    #[cfg(target_os = "linux")]
+                    if io_error.raw_os_error() == Some(38) {
+                        return Ok(WatcherFallback::Fallback(PollWatcher::new(event_handler, config)?));
+                    }
+                },
+                _ => (),
+            }
+            return Err(e);
+        },
+    }
+}
+
+/// Wrapper for a fallback initialized watcher.
+/// 
+/// See [`recommended_watcher_fallback`](recommended_watcher_fallback)
+#[derive(Debug)]
+pub enum WatcherFallback {
+    /// Native watcher, no error occured
+    Native(RecommendedWatcher),
+    /// Fallback watcher, known platform issue occured
+    /// 
+    /// For example the Docker with Linux on MacOS M1 bug
+    Fallback(PollWatcher)
+}
+
+impl WatcherFallback {
+    /// Returns the watcher inside
+    pub fn take_boxed(self) -> Box<dyn Watcher>{
```

Comment:
personally I would prefer that this implemented `watch`. Is that very verbose/difficult?

Reply:
that's a very good idea! and it would make the enum more worthwhile (because otherwise I don't actually see much of a value over directly boxing it, except for claiming the allocation-free throne, while not being useful at all)

Reply:
Ok I tried this.. And it gets really awkward.

- We can't implement `Watcher::new` due to missing `Copy`/`Clone `on `event_handler`.
- We can't implement a real `kind()`, as we don't have access to `self`. This kind of API is specifically not designed for runtime-selected watchers it seems. We could still implement the functions without the trait, or leave them `unimplemented!()` for the new and the additional watcher-kind as in my new code.

See also the added comments.